testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSystemTableCompatibility.java

```diff
@@ -92,4 +92,65 @@ public void testTablePropertiesWithTableFeatures()
+
+        List<Row> expectedBeforeDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 3L, 1347L, "{\"b\":{\"min\":11,\"max\":19,\"null_count\":0}}"),
+                row("{\"a\":1,\"c\":\"d\"}", 1L, 449L, "{\"b\":{\"min\":13,\"max\":13,\"null_count\":0}}"));
+
+        List<Row> expectedAfterFirstDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 2L, 898L, "{\"b\":{\"min\":11,\"max\":17,\"null_count\":0}}"),
+                row("{\"a\":1,\"c\":\"d\"}", 1L, 449L, "{\"b\":{\"min\":13,\"max\":13,\"null_count\":0}}"));
+
+        List<Row> expectedAfterSecondDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 2L, 898L, "{\"b\":{\"min\":11,\"max\":17,\"null_count\":0}}"));
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 11, 'a'), (1, 19, 'a'), (1, 17, 'a'), (1, 13, 'd')");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedBeforeDelete);
+            onDelta().executeQuery("DELETE FROM default." + tableName + " WHERE b = 19");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedAfterFirstDelete);
+            onDelta().executeQuery("DELETE FROM default." + tableName + " WHERE c = 'd'");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedAfterSecondDelete);
+        }
+        finally {
+            onDelta().executeQuery("DROP TABLE IF EXISTS default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testTablePartitionsWithNoColumnStats()
+    {
+        String tableName = "test_table_partitions_with_no_column_stats_" + randomNameSuffix();
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "(a INT, b INT, c VARCHAR(1)) " +
+                "USING delta " +
+                "PARTITIONED BY (a, c) " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                "TBLPROPERTIES ('delta.dataSkippingNumIndexedCols' = 0)");
+
+        List<Row> expected = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 3L, 1347L, "{\"b\":{\"min\":null,\"max\":null,\"null_count\":0}}"),
+                row("{\"a\":1,\"c\":\"d\"}", 1L, 449L, "{\"b\":{\"min\":null,\"max\":null,\"null_count\":0}}"));
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 11, 'a'), (1, 19, 'a'), (1, 17, 'a'), (1, 13, 'd')");
```

Comment:
nit: Let's avoid 1 character on the 3rd column. It's confusing for me since it's same as the column name.